### DISCUSSION TITLE
Hotreloading extensions

### DISCUSSION THREAD

**Discussion by Fusseldieb** [directus/directus#5938] [Feature Requests]
Hi,

I am developing custom displays for my Directus dev instance, but everytime I want to see a change, I need to do `Ctrl^C` to stop the server and `npx directus start` to start it again.

It would be awesome to include a command like "npx directus hotreload" which listens for changes in the `extensions/` folder and reloads the server. This would already simplify the workflow.

Related #5539 #3017

In the future it would be awesome to see the change without reloading the entire page, but that's probably a bit harder to do.

**Comment by Fusseldieb** [comment]
In the meantime I'm running it with nodemon, does the same

Inside the custom display dev folder I added a script into package.json that upon `build`ing does a `cp` and copies the resulting `index.js` from the rollup into the Directus dev instance. With the command below, it notices the change and automatically restarts.

`npx nodemon --exec "npx directus start"`

```
$ npx nodemon --exec "npx directus start"
[nodemon] 2.0.7
[nodemon] to restart at any time, enter `rs`
[nodemon] watching path(s): *.*
[nodemon] watching extensions: js,mjs,json
[nodemon] starting `npx directus start`
11:47:58 Ô£¿ Server started at port 80
[nodemon] restarting due to changes...
[nodemon] starting `npx directus start`
11:48:17 Ô£¿ Server started at port 80
```

**Comment by azrikahar** [comment]
Update: This has been implemented in #10881 with the addition of [`EXTENSIONS_AUTO_RELOAD` environment variable](https://docs.directus.io/configuration/config-options/#extensions).

Though do note that App extensions still requires a refresh of the browser for the changes to reflect properly 👍
